src/components/MainContent/MainContent.tsx

```diff
@@ -0,0 +1,266 @@
+    const isTodoUnique = todos.filter(todo => (
+      todo.title === title
+    )).length === 0;
+
+    if (isValidTempTodo && isTodoUnique) {
+      const newTempTodo = {
+        userId: user.id,
+        title,
+        completed: false,
+      };
+
+      setIsTodoAdding(true);
+      setTempTodo(newTempTodo);
+      setTodos(currentTodos => [...currentTodos, { ...newTempTodo, id: 0 }]);
+      addOnLoadingTodo(0);
+    } else {
+      setError('Can\'t add new todo');
+    }
+  };
+
+  // Server actions
+  const loadTodosFromServer = async () => {
+    try {
+      setTodos(await getTodos(user.id));
+      // eslint-disable-next-line
+    } catch (error: any) {
+      // eslint-disable-next-line no-console
+      console.log('error', error.message);
+      setError('Can\'t load todos');
+    }
+  };
+
+  const addNewTodoOnServer = async (newTodo: TodoForServer) => {
+    try {
+      await addTodo(user.id, newTodo);
+      await loadTodosFromServer();
+      setTempTodo(null);
+      // eslint-disable-next-line
+    } catch (error: any) {
+      // eslint-disable-next-line no-console
+      console.log('error', error.message);
+      setError('Unable to add a todo');
+    }
+  };
+
+  const deleteTodoOnServer = async (id: number) => {
+    try {
+      addOnLoadingTodo(id);
+      await deleteTodo(id);
+      await loadTodosFromServer();
+      // eslint-disable-next-line
+    } catch (error: any) {
+      // eslint-disable-next-line no-console
+      console.log('error', error.message);
+      setError('Unable to delete a todo');
+    }
+  };
+
+  const changeTodoOnServer = async (
+    id: number,
+    todo: Partial<Todo>,
+  ) => {
+    try {
+      addOnLoadingTodo(id);
+      await editTodo(id, todo);
+      await loadTodosFromServer();
+      deleteFromLoadingTodo(id);
+      // eslint-disable-next-line
+    } catch (error: any) {
+      // eslint-disable-next-line no-console
+      console.log('error', error.message);
+      setError('Unable to update a todo');
+    }
+  };
+
+  const changeCompletedForGroup = async (group: Todo[], value: boolean) => {
+    group.map(todo => addOnLoadingTodo(todo.id));
+    await Promise.all(
+      todos.map(todo => editTodo(todo.id, { completed: value })),
+    );
+    await loadTodosFromServer();
+    todos.map(todo => deleteFromLoadingTodo(todo.id));
+  };
+
+  const setAllTodosCompletedOnServer = async () => {
+    try {
+      if (isAllTodosCompleted) {
+        await changeCompletedForGroup(completedTodos, false);
+      } else {
+        await changeCompletedForGroup(activeTodos, true);
+      }
+      // eslint-disable-next-line
+    } catch (error: any) {
+      // eslint-disable-next-line no-console
+      console.log('error', error.message);
+      setError('Can\'t change todos');
+    }
+  };
+
+  const deleteAllCompletedTodos = async () => {
+    try {
+      await Promise.all(
+        completedTodos.map(todo => deleteTodoOnServer(todo.id)),
```

Comment:
use forEach

Reply:
I can't use forEach in Promise.all cause I need new array